runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1554,37 +1553,18 @@ static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node,
    // Init size and mustBeZero ('0') fields to 0
    generateMemImmInstruction(TR::InstOpCode::S4MemImm4, node, generateX86MemoryReference(targetReg, fej9->getOffsetOfContiguousArraySizeField(), cg), 0, cg);
    generateMemImmInstruction(TR::InstOpCode::S4MemImm4, node, generateX86MemoryReference(targetReg, fej9->getOffsetOfDiscontiguousArraySizeField(), cg), 0, cg);
+   if (TR::Compiler->om.compressObjectReferences())
+      { // Clear padding in contiguous array header layout. +4 because size field is 4 bytes wide.
+      generateMemImmInstruction(TR::InstOpCode::S4MemImm4, node, generateX86MemoryReference(targetReg, fej9->getOffsetOfDiscontiguousArraySizeField() + 4, cg), 0, cg);
```

Comment:
Looking at [`J9IndexableObjectDiscontiguous`](https://github.com/eclipse-openj9/openj9/blob/01b6a394d2850c9ea080068b2f5484576bdcd397/runtime/oti/j9nonbuilder.h#L3203-L3210), if I'm reading it correctly, I think the `padding` field is present with compressed references enabled or in 32-bit mode.  If `padding` needs to be cleared if compressed references are used does it need to be cleared in 32-bit mode as well.

Sorry if I'm misreading this!  I haven't really needed to dig into object header layout much.